Framework/PythonInterface/plugins/algorithms/LoadAndMerge.py

```diff
@@ -199,5 +201,19 @@ def PyExec(self):
 
         self.setProperty("OutputWorkspace", mtd[output])
 
+        if not self.getAlwaysStoreInADS():
+            self.remove_output_from_ads(output)
+
+    def remove_output_from_ads(self, output_ws_name):
+        remove_ws = [output_ws_name]
+        if self.is_group_ws(output_ws_name):
+            remove_ws.extend(list(mtd[output_ws_name].getNames()))
+        for ws in remove_ws:
+            mtd.remove(ws)
+
+    @staticmethod
+    def is_group_ws(ws_name):
+        return issubclass(type(mtd[ws_name]), WorkspaceGroup)
```

Comment:
I'm not sure if this method is necessary. Given we're already accessing the workspace object through `mtd`, is there a reason not to use `mtd[ws_name].isGroup()`?

Reply:
I don't think so - I just copied this function from elsewhere in Reflectometry.